inc/hcupc-support.h

```diff
@@ -103,6 +103,17 @@ inline void execute_hcupc_lambda(T* lambda) {
 			 * If a lambda is heap allocated then the pointer will not be valid at
 			 * remote place. Due to this we have to first copy the content of this
 			 * lambda into a stack allocated lambda.
+			 *
+			 * XXX - This description doesn't make any sense to me.
+			 * There's no way in C++ to differentiate a pointer to
+			 * an object with dynamic vs automatic storage duration,
+			 * so I don't see how this could be true. To further
+			 * complicate this, we're currently using heap-allocated
+			 * stacks for our fibers, so everything is in the heap
+			 * (i.e., even address-range-checking wouldn't work for
+			 * differentiating between stack and heap). This will
+			 * need to be refactored later, but I need to find out
+			 * what exactly it's supposed to be doing first.
 			 */
 
 			/*
```

Comment:
Ok, let me try to think.. When a task (lambda) is popped from deque its a void* pointer. The asyncAny interface specifies a task that can be either executed by a local worker or a remote worker (distributed work-stealing). In case of local execution, the above comment doesn't make sense. However, if a remote worker has to steal this task (which is a void* pointer on the deque of remote computation worker), it can only be done if the type of lambda can be derived after the task is popped from the deque. We have to pass the content of lambda and not the pointer to this lambda at remote place. Thats why I was saving the "type" of lambda in the task structure which would help in doing memcpy of this lambda into a stack allocated lambda whenever a distributed work-stealing is happening.

However, now as entire support for HCUPC++ is broken, I am not sure if we really need this now.